### ISSUE TITLE
Segfault when using the host SMB sharefolder on Windows (from Linux client)

### ISSUE THREAD

**Issue by pan1nx** [FreeRDP/Remmina#1157]
* Client (OS name and version): Ubuntu 14.04.5 LTS x64
* Remmina version (remmina --version): 1.2.0-rcgit-18 (git rcgit-18)
* Desktop environment (GNOME, Unity, KDE, ..): KDE
* Connecting to (OS and version): Microsoft Windows [Version 10.0.14393]
* Connecting via (RDP, VNC, ...): RDP
* RDP issues only, freerdp version (xfreerdp --version): version 2.0.0~dev from 2.0.0~dev201703030839+dfsg201704061532-0ubuntu0~git9fd3974+e33fbeb~ubuntu14.04.1

When in windows, if trying to mount/access the tsclient SMB connection/mount I get a segmentation fault. Here is the dump:
``` bash
Program received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7fff9f53c700 (LWP 10520)]
0x00007fffe77b220b in FindClose () from /usr/lib/x86_64-linux-gnu/libwinpr2.so.2
(gdb) bt
#0  0x00007fffe77b220b in FindClose () from /usr/lib/x86_64-linux-gnu/libwinpr2.so.2
#1  0x00007fffe77b26b7 in GetFileAttributesExA () from /usr/lib/x86_64-linux-gnu/libwinpr2.so.2
#2  0x00007fffe77b275b in GetFileAttributesExW () from /usr/lib/x86_64-linux-gnu/libwinpr2.so.2
#3  0x00007fffe002b886 in ?? () from /usr/lib/x86_64-linux-gnu/freerdp2/libdrive-client.so
#4  0x00007fffe77e96c6 in ?? () from /usr/lib/x86_64-linux-gnu/libwinpr2.so.2
#5  0x00007ffff5fe9184 in start_thread (arg=0x7fff9f53c700) at pthread_create.c:312
#6  0x00007ffff4fbcbed in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:111
```

This used to work one week ago. I am using the Ubuntu PPA/ remmina-ppa-team/remmina-next which upgraded last Friday the packages and after that I started having issues with the sharefolder.

Thanks,
Peter

**Comment by giox069** [comment]
Thank you for reporting. Yes, we updated libfreerdp in the PPA few days ago, and we have a regression bug. Can you better explain "trying to mount/access the tsclient SMB mount" ? Are you using the "Share folder" option on the remmina profile editor? Can you check if you have non-ascii characters on the folder path?

**Comment by giox069** [comment]
Never mind. We are able to reproduce the problem. I hope we will find a fix as soon as possible.
Thank you for reporting.

**Comment by pan1nx** [comment]
Hi, Nope. No non-ascii characters on the folder path. Just /home/user...

Here is with the today's build:
Program received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7fff9f53c700 (LWP 11721)]
0x00007fffe77b220b in FindClose () from /usr/lib/x86_64-linux-gnu/libwinpr2.so.2
(gdb) bt
#0  0x00007fffe77b220b in FindClose () from /usr/lib/x86_64-linux-gnu/libwinpr2.so.2
#1  0x00007fffe77b26b7 in GetFileAttributesExA () from /usr/lib/x86_64-linux-gnu/libwinpr2.so.2
#2  0x00007fffe77b275b in GetFileAttributesExW () from /usr/lib/x86_64-linux-gnu/libwinpr2.so.2
#3  0x00007fffe002b886 in ?? () from /usr/lib/x86_64-linux-gnu/freerdp2/libdrive-client.so
#4  0x00007fffe77e96c6 in ?? () from /usr/lib/x86_64-linux-gnu/libwinpr2.so.2
#5  0x00007ffff5fe9184 in start_thread (arg=0x7fff9f53c700) at pthread_create.c:312
#6  0x00007ffff4fbcbed in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:111

I can install the debug symbols and find you exactly what causes it if you want... Just let me know.

Cheers,
Peter

**Comment by giox069** [comment]
@pan1nx: it's a problem on libfreerdp. There is already a patch ready in their PR https://github.com/FreeRDP/FreeRDP/pull/3902, we just need to wait when they will merge the pull request and then we will update the PPA as soon as possible.

**Comment by pan1nx** [comment]
Ok. Cool. I have been testing it and it looks like I could browse the directory tree from the shared folder but not the content (meaning, if I open a directory I crash). Thanks anyway @giox069 for your kind and fast response.

**Comment by giox069** [comment]
The remmina-next PPA has been updated with the patched libfreerdp. You can update your ubuntu and test it.

**Comment by pan1nx** [comment]
Hi,

Yes. The latest update fixed it. For the record it is fixed now in "2.0.0~dev201703030839+dfsg201704111144-0ubuntu0~git59a8418+e33fbeb~ubuntu14.04.1" update.

Closing the issue. Thanks again @giox069 for the nice response.
